endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpQueryParamHeaderValidator.java

```diff
@@ -38,49 +42,62 @@ public void validateHeader(String name, Object received, Object control, TestCon
         if (ValidationMatcherUtils.isValidationMatcherExpression(Optional.ofNullable(control)
                 .map(Object::toString)
                 .orElse(""))) {
-            super.validateHeader(HttpMessageHeaders.HTTP_QUERY_PARAMS, received, control, context, validationContext);
+            super.validateHeader(HTTP_QUERY_PARAMS, received, control, context, validationContext);
             return;
         }
 
-        Map<String, String> receiveParams = convertToMap(received);
-        Map<String, String> controlParams = convertToMap(control);
+        // TODO Christoph Deppisch: I changed this to support multi lists, e.q. required for array query parameters.
+        //  Not sure about consequences though. Therefore, i call a new method super.validateHeaderArray below.
+        //  Maybe we should fix this in general.
```

Comment:
shoutout to @christophd, did you see (and approve of) this?

Reply:
Honestly, I am not sure about the consequences neither. All existing tests are green so that drives my confidence

Reply:
love that mindset/confidence 🥇